src/App.js

```diff
@@ -7,29 +7,39 @@ import axios from 'axios';
 
 class App extends React.Component {
 
-  constructor(props){
+  constructor(props) {
     super(props);
     this.state = [
-      {name: "Breakfast"},
-      {name: "Lunch"},
-      {name: "Dinner"},
-      {name: "Sides"},
-      {name: "Dessert"}
+      { name: "Breakfast", Menu: "" },
+      { name: "Lunch", Menu: "" },
+      { name: "Dinner", Menu: "" },
+      { name: "Sides", Menu: "" },
+      { name: "Dessert", Menu: "" }
     ]
   }
-  
-   componentDidMount(){
-
-     
-     this.state.map(MenuItems => {
-       axios.get('https://entree-f18.herokuapp.com/v1/menu/12')
-       .then (items => {
-         localStorage.setItem(MenuItems.name, JSON.stringify(items.data.menu_items))
-       })
-      ;
-     })
-     
-   
+
+  componentDidMount() {
+
+    // Check if menu data already exists in local storage.
+    //If so, store data in local storage in menu state, else pull from API
+
+    if (localStorage.getItem('Breakfast')) {
+      this.state.map(MenuItems_Local => {
+        this.state.Menu = JSON.parse(localStorage.getItem(MenuItems_Local.name));
+      })
+    }
+    else {
+      this.state.map(MenuItems => {
+        axios.get('https://entree-f18.herokuapp.com/v1/menu/12')
+          .then(items => {
+            localStorage.setItem(MenuItems.name, JSON.stringify(items.data.menu_items))
+            this.state.Menu = items.data.menu_items;
+          })
+          ;
+      })
+
+
```

Comment:
You could probably get rid of the empty lines on 40-41

Reply:
Done, Lines 40 and 41 deleted